main.go

```diff
@@ -105,17 +101,10 @@ func main() {
 		return
 	}
 
-	err := execWithCredentials(args, creds)
+	err = execWithCredentials(args, creds)
 	must(err)
 }
 
-func cleanEnv() {
```

Comment:
The env cleaning is ultimately what prevents the nested calls from working today. Once you remove the env clean before the sdk's assume role or session creation calls, it becomes unnecessary _after_ those calls because we're about to overwrite the values with the newly assumed role anyway.